drivers/neopixel/neopixel.py

```diff
@@ -1,44 +1,44 @@
 # NeoPixel driver for MicroPython
 # MIT license; Copyright (c) 2016 Damien P. George, 2021 Jim Mussared
 
-from micropython import const
 from machine import bitstream
 
-_BITSTREAM_TYPE_HIGH_LOW = const(0)
-_TIMING_WS2818_800 = (400, 850, 800, 450)
-_TIMING_WS2818_400 = (800, 1700, 1600, 900)
-
 
 class NeoPixel:
+    # G R B W
     ORDER = (1, 0, 2, 3)
 
-    def __init__(self, pin, n, bpp=3, timing=1):
+    def __init__(self, pin, n, bpp=3, t=1):
         self.pin = pin
         self.n = n
         self.bpp = bpp
         self.buf = bytearray(n * bpp)
         self.pin.init(pin.OUT)
-        self.timing = (
-            (_TIMING_WS2818_800 if timing else _TIMING_WS2818_400)
-            if isinstance(timing, int)
-            else timing
+        # Timing arg can either be 1 for 800kHz or 0 for 400kHz,
+        # or a user-specified timing ns tuple (low_0, high_0, low_1, high_1).
+        self.t = (
+            ((400, 850, 800, 450) if t else (800, 1700, 1600, 900)) if isinstance(t, int) else t
```

Comment:
This will actually construct the 4-tuple each time a new NeoPixel object is created.  But there's a tradeoff whether it's globally computed or not, and I'm not sure what to optimise for here.

Reply:
Eventually the compiler will be good enough such that the tuples can be true constant objects, but that's not currently the case.

Reply:
I think optimising for bytecode makes sense here and when they can become true constant objects we can update it.